packages/storybook-builder-vite/source-loader-plugin.js

```diff
@@ -1,24 +1,18 @@
-const inject =
-    require('@storybook/source-loader/dist/cjs/abstract-syntax-tree/inject-decorator').default;
+const transform =
+    require('@storybook/source-loader').default;
 
 module.exports.sourceLoaderPlugin = function () {
     return {
         name: 'storybook-vite-source-loader-plugin',
         enforce: 'pre',
-        transform(src, id) {
+        async transform(src, id) {
             if (id.match(/\.stories\.[jt]sx?$/)) {
-                const { source, sourceJson, addsMap } = inject(src, id);
-                const preamble = `
-                    /* eslint-disable */
-                    // @ts-nocheck
-                    // @ts-ignore
-                    var __STORY__ = ${sourceJson};
-                    // @ts-ignore
-                    var __LOCATIONS_MAP__ = ${JSON.stringify(addsMap)};
-                `;
+                // We need to mock 'this' when calling transform from @storybook/source-loader
+                const mockClassLoader = { emitWarning: () => {}, resourcePath: id };
+                const code = await transform.call(mockClassLoader, src);
```

Comment:
Any idea what this line is for? https://github.com/storybookjs/storybook/blob/c67c22afa75728eb00d782e88b4d43645d3a3253/lib/source-loader/src/dependencies-lookup/readAsObject.js#L12

Do we also need to mock out `extension` or whatever `getOptions` is looking for?  My guess is "no", especially since this is working as-is.

Reply:
I don't think we do, it's webpack specific loader stuff and if options.parser or classLoader.extension is not defined the filename is.